src/main/java/graphql/nadel/engine/transformation/FieldUtils.java

```diff
@@ -135,4 +137,11 @@ public static ExecutionResultNode mapChildren(ExecutionResultNode executionResul
         return executionResultNode.withNewChildren(newChildren);
     }
 
+    public static String randomNodeId() {
+        return String.valueOf(RANDOM.nextInt(MAX_VALUE));
+    }
+
+    public static String randomFieldSuffix() {
+        return String.valueOf(RANDOM.nextInt(MAX_VALUE));
```

Comment:
My only concern with these is that there might be collisions? But I imagine the possibility should be low.

Note: if you provide an upper bound it only generates 0 to that upper bound (no negative numbers).

Reply:
we are only worried about a collision insdie a single request.

How do we know that new UUID is slow - did it turn up on some profiling?  Do we benchmark it?

Reply:
Nah I just saw it and thought it'd be slow. I did some benchmarking, and it seems to be much much slower.

I'm not too familiar with the benchmarking but I took [Benchmark.java](https://github.com/graphql-java/graphql-java/blob/000eb5274793b7b41f863d3d96afcbb4e05646f8/src/test/java/benchmark/BenchMark.java) from GraphQL Java and changed the benchmark methods call the following instead of `executeQuery`:

```java
    private static int test = 0;

    void uuid() {
        String pathSeg = "";
        String fieldId = "new-field_" + pathSeg + "_" + UUID.randomUUID().toString();
        test += fieldId.hashCode() > 0 ? 1 : 0;
    }
```

And:

```java
    private static int test = 0;

    private static final Random RANDOM = new Random();

    void uuid() {
        String pathSeg = "";
        String fieldId = "new-field_" + pathSeg + "_" + String.valueOf(RANDOM.nextInt(MAX_VALUE));
        test += fieldId.hashCode() > 0 ? 1 : 0;
    }
```

The results on my laptop:

UUID:

```
Benchmark                                    Mode  Cnt       Score      Error  Units
BenchMark.benchMarkSimpleQueriesThroughput  thrpt   15  390936.630 ± 6848.148  ops/s
BenchMark.benchMarkSimpleQueriesAvgTime      avgt   15       0.003 ±    0.001  ms/op
```

Random Number ToString:

```
Benchmark                                    Mode  Cnt        Score        Error  Units
BenchMark.benchMarkSimpleQueriesThroughput  thrpt   15  3348973.006 ± 300036.848  ops/s
BenchMark.benchMarkSimpleQueriesAvgTime      avgt   15       ≈ 10⁻⁴               ms/op
```

Reply:
So definitely faster - problem of course is that you need ENOUGH of the UUIDS to be generated to ACTUALLY make any savings.

if we have 1000s of UUIDS created per query  then real savings can be made - but if its 10s - then it wont be noticed.  Faster but not noticeably faster

I am happy to use this but it wont be faster in practice I think